Testscripts/Windows/Power-Hibernate.ps1

```diff
@@ -0,0 +1,213 @@
+				Write-LogInfo "Copied the script files to the VM"
+		}
+		#endregion
+
+		# Configuration for the hibernation
+		Run-LinuxCmd -ip $AllVMData.PublicIP -port $AllVMData.SSHPort -username $user -password $password -command "./SetupHbKernel.sh" -RunInBackground -runAsSudo -ignoreLinuxExitCode:$true | Out-Null
+		Write-LogInfo "Executed SetupHbKernel script inside VM"
+
+		# Wait for kernel compilation completion. 60 min timeout
+		$timeout = New-Timespan -Minutes 60
+		$sw = [diagnostics.stopwatch]::StartNew()
+		while ($sw.elapsed -lt $timeout){
+			$vmCount = $AllVMData.Count
+			Wait-Time -seconds 15
+			$state = Run-LinuxCmd -ip $VMData.PublicIP -port $VMData.SSHPort -username $user -password $password "cat ~/state.txt"
+			if ($state -eq "TestCompleted") {
+				$kernelCompileCompleted = Run-LinuxCmd -ip $VMData.PublicIP -port $VMData.SSHPort -username $user -password $password "cat ~/constants.sh | grep setup_completed=0"
+				if ($kernelCompileCompleted -ne "setup_completed=0") {
+					Write-LogErr "SetupHbKernel.sh run finished on $($VMData.RoleName) but setup was not successful!"
+				} else {
+					Write-LogInfo "SetupHbKernel.sh finished on $($VMData.RoleName)"
+					$vmCount--
+				}
+				break
+			} elseif ($state -eq "TestSkipped") {
+				Write-LogInfo "SetupHbKernel.sh finished with SKIPPED state!"
+				$testResult = $resultSkipped
+				return $testResult
+			} elseif (($state -eq "TestFailed") -or ($state -eq "TestAborted")) {
+				Write-LogErr "SetupHbKernel.sh didn't finish successfully!"
+				$testResult = $resultAborted
+				return $testResult
+			} else {
+				Write-LogInfo "SetupHbKernel.sh is still running in the VM!"
+			}
+		}
+		if ($vmCount -le 0){
+			Write-LogInfo "SetupHbKernel.sh is done"
+		} else {
+			Throw "SetupHbKernel.sh didn't finish in the VM!"
+		}
+
+		# Reboot VM to apply swap setup changes
+		Write-LogInfo "Rebooting All VMs!"
+		$TestProvider.RestartAllDeployments($AllVMData)
+
+		# Check the VM status before hibernation
+		$vmStatus = Get-AzVM -Name $vmName -ResourceGroupName $rgName -Status
+		if ($vmStatus.Statuses[1].DisplayStatus = "VM running") {
+			Write-LogInfo "$vmStatus.Statuses[1].DisplayStatus: Verified successfully VM status is running before hibernation"
+		} else {
+			Write-LogErr "$vmStatus.Statuses[1].DisplayStatus: Could not find the VM status before hibernation"
+			throw "Can not identify VM status before hibernate"
+		}
+
+		# Hibernate the VM
+		Run-LinuxCmd -ip $AllVMData.PublicIP -port $AllVMData.SSHPort -username $user -password $password -command "sudo ~/test.sh" -ignoreLinuxExitCode:$true | Out-Null
+		Write-LogInfo "Sent hibernate command to the VM"
+		Start-Sleep -s 120
+
+		# Verify the VM status
+		# Can not find if VM hibernation completion or not as soon as it disconnects the network. Assume it is in timeout.
+		$vmStatus = Get-AzVM -Name $vmName -ResourceGroupName $rgName -Status
+		if ($vmStatus.Statuses[1].DisplayStatus = "VM stopped") {
```

Comment:
if ($vmStatus.Statuses[-1].DisplayStatus -eq "VM stopped") {
here = is assigned "VM stopped" to $vmStatus.Statuses[1].DisplayStatus